File: app/src/main/java/io/tanjundang/study/test/animation/BtnAnimation.java
package io.tanjundang.study.test.animation;

import android.view.View;
import android.view.animation.Animation;
import android.view.animation.Transformation;

/**
 * Developer: TanJunDang
 * Email: TanJunDang324@gmail.com
 * Date: 2016/8/26
 */

public class BtnAnimation extends Animation {

    View view;
    int width;
    int widthto;

    public BtnAnimation(View view) {
        this.view = view;
        width = view.getWidth();
        if (width == view.getHeight()) {
            widthto = 4 * width;
        } else {
            widthto = view.getHeight();
        }
    }

    @Override
    protected void applyTransformation(float interpolatedTime, Transformation t) {
        super.applyTransformation(interpolatedTime, t);
        view.getLayoutParams().width = width + (int) ((widthto - width) * interpolatedTime);
        view.requestLayout();
    }


}

File: app/src/main/java/io/tanjundang/study/knowledge/animation/BtnAnimation.kt
package io.tanjundang.study.knowledge.animation

import android.view.View
import android.view.animation.Animation
import android.view.animation.Transformation

/**
 * Developer: TanJunDang
 * Email: TanJunDang324@gmail.com
 * Date: 2016/8/26
 */

class BtnAnimation(internal var view: View) : Animation() {
    internal var width: Int = 0
    internal var widthto: Int = 0

    init {
        width = view.width
        if (width == view.height) {
            widthto = 4 * width
        } else {
            widthto = view.height
        }
    }

    override fun applyTransformation(interpolatedTime: Float, t: Transformation) {
        super.applyTransformation(interpolatedTime, t)
        view.layoutParams.width = width + ((widthto - width) * interpolatedTime).toInt()
        view.requestLayout()
    }


}
